extensions/process/mod-process.c

```diff
@@ -190,7 +190,11 @@ DECLARE_NATIVE(get_os_browsers)
         "]"
     );
 
-  #else // Just try /usr/bin/open on POSIX, OS X, Haiku, etc.
+  #elif TO_HAIKU
+
+    rebElide("append", list, rebT("open %1"));
```

Comment:
Glad you encountered the user API, it's kind of what kicked off Ren-C as a project:

  https://forum.rebol.info/t/limiting-api-entry-points-in-favor-of-exchanging-strings/540

I think how this is accomplished is pretty neat (based on making sure all value cell and series don't have a first byte that could be from a legal UTF-8 string).  It makes bridging with C literate, for a complex example see things like the ODBC extension:

  https://github.com/metaeducation/ren-c/blob/8a0da5521bf438c0279561c726fa99ecf079cf11/extensions/odbc/mod-odbc.c#L539

The way this stuff was done in R3-Alpha was very...unfit.